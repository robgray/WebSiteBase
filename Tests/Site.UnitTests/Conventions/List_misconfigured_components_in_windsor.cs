using Castle.MicroKernel;
using Castle.Windsor;
using Castle.Windsor.Diagnostics;
using WebBase.Mvc.App_Start;
using log4net;
using log4net.Config;

namespace Mvc.Tests.Conventions
{
    public class List_misconfigured_components_in_Windsor :
       WindsorConventionTest<IPotentiallyMisconfiguredComponentsDiagnostic>
    {
        protected override WindsorConventionData<IHandler> SetUp()
        {
            var container = CreateAndConfigureContainerLikeInTheApp();
            return
                new WindsorConventionData<IHandler>(container)
                {
                    FailDescription = "The following components come up as potentially unresolvable",
                    FailItemDescription = MisconfiguredComponentDescription
                }.WithApprovedExceptions("it's *potentially* for a reason");
        }

        private IWindsorContainer CreateAndConfigureContainerLikeInTheApp()
        {
            XmlConfigurator.Configure();

            var appLogger = LogManager.GetLogger(GetType());
            AssembliesConfig.LookInBinSubfolder = false; // NOTE: this is temporary until we remove that code completely
            var modules = AssembliesConfig.Configure(appLogger);
            var container = ContainerConfig.Configure(appLogger);


            return container;
        }
    }
}
