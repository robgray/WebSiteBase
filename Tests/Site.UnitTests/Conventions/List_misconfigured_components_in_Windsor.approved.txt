The following components come up as potentially unresolvable
	